lib/internal/errors.js

```diff
@@ -86,6 +87,51 @@ function makeNodeError(Base) {
   };
 }
 
+function makeNodeError2(Base, key) {
+  return class NodeError extends Base {
+    constructor(...args) {
+      super(message(key, args));
+    }
+
+    get name() {
+      return `${super.name} [${key}]`;
+    }
+
+    set name(value) {
+      defineProperty(this, 'name', {
+        configurable: true,
+        enumerable: true,
+        value,
+        writable: true
+      });
+    }
+
+    get code() {
+      return key;
+    }
+
+    set code(value) {
```

Comment:
is there any use case for this?

Reply:
My guess is that userland modules (not aware of the standardized node core error objects) may have already been setting this property to something else.

Reply:
Yes, this was specifically asked for by userland module authors who need the ability to keep specifying their own codes.